Lib/_weakrefset.py

```diff
@@ -51,10 +51,15 @@ def _remove(item, selfref=ref(self)):
             self.update(data)
 
     def _commit_removals(self):
-        l = self._pending_removals
+        pop = self._pending_removals.pop
         discard = self.data.discard
-        while l:
-            discard(l.pop())
```

Comment:
the race is between the `while l:` and the `l.pop()` there can be one pending removal in the list, `l`, and two threads passing the `while l:` guard

This PR uses EAFP to avoid that scenario